src/elexsolver/QuantileRegressionSolver.py

```diff
@@ -49,21 +49,30 @@ def _check_matrix_condition(self, x):
             return False
         return True
 
-    def __solve(self, x, y, weights, verbose):
+    def get_loss_function(self, x, y, coefficients, weights):
+        y_hat = x @ coefficients
+        residual = y - y_hat
+        return cp.sum(cp.multiply(weights, 0.5 * cp.abs(residual) + (self.tau.value - 0.5) * residual))
+
+    def get_regularizer(self, coefficients):
+        return cp.pnorm(coefficients, p=2) ** 2
+
+    def __solve(self, x, y, weights, lambda_, verbose):
         """
         Sets up the optimization problem and solves it
         """
         self._check_matrix_condition(x)
         coefficients = cp.Variable((x.shape[1],))
-        y_hat = x @ coefficients
-        residual = y - y_hat
-        loss_function = cp.sum(cp.multiply(weights, 0.5 * cp.abs(residual) + (self.tau.value - 0.5) * residual))
+        loss_function = self.get_loss_function(x, y, coefficients, weights)
+        loss_function += lambda_ * self.get_regularizer(coefficients)
         objective = cp.Minimize(loss_function)
         problem = cp.Problem(objective)
         problem.solve(solver=self.solver, verbose=verbose, **self.KWARGS.get(self.solver, {}))
         return coefficients, problem
 
-    def fit(self, x, y, tau_value=0.5, weights=None, verbose=False, save_problem=False, normalize_weights=True):
+    def fit(
+        self, x, y, tau_value=0.5, weights=None, lambda_=0, verbose=False, save_problem=False, normalize_weights=True
```

Comment:
Are you opening a corresponding model PR where the new lambda var is passed up into the fit-model functions and ultimately as a user input / kwarg?

Reply:
yup, working on that PR right now!